src/hardware/ipxserver.cpp

```diff
@@ -212,10 +220,18 @@ bool IPX_StartServer(uint16_t portnum)
 		ipxServerSocket = SDLNet_UDP_Open(portnum);
 		if(!ipxServerSocket) return false;
 
-		for (uint16_t i = 0; i < SOCKETTABLESIZE; ++i)
-			connBuffer[i].connected = false;
+		for (auto& i : connBuffer) {
+			i.connected = false;
+		}
 
-		TIMER_AddTickHandler(&IPX_ServerLoop);
+		ipx_server_running = true;
+		ipx_server_thread  = std::thread([]() {
```

Comment:
This will throw an exception if a caller calls `IPX_StartServer` twice as the thread will be destroyed without having been joined first.

Often I will keep a worker thread running for the duration of the entire application waiting on a start event to avoid the overhead of repeatedly creating/destroying threads.  If `IPX_StartServer` gets called infrequently (I'm guessing it does?), it's fine to spin up a thread on demand but you might want to explicitly call `IPX_StopServer` beforehand.

Reply:
good point - it's pretty safe, since the user explicitly has to do `ipxnet startserver`, so I don't want a thread running if the user doesn't want it, however it's easy to put a quick check for `!ipx_server_running` before spinning it up, just in case.